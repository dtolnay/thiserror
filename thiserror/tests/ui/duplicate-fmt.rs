use thiserror::Error;

#[derive(Error, Debug)]
#[error("...")]
#[error("...")]
pub struct Error;

#[derive(Error, Debug)]
#[error(fmt = core::fmt::Octal::fmt)]
#[error(fmt = core::fmt::LowerHex::fmt)]
pub enum FmtFmt {}

#[derive(Error, Debug)]
#[error(fmt = core::fmt::Octal::fmt)]
#[error(transparent)]
pub enum FmtTransparent {}

#[derive(Error, Debug)]
#[error(fmt = core::fmt::Octal::fmt)]
#[error("...")]
pub enum FmtDisplay {}

fn main() {}
