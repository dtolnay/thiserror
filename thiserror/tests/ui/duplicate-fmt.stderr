error: only one #[error(...)] attribute is allowed
 --> tests/ui/duplicate-fmt.rs:5:1
  |
5 | #[error("...")]
  | ^^^^^^^^^^^^^^^

error: duplicate #[error(fmt = ...)] attribute
  --> tests/ui/duplicate-fmt.rs:10:1
   |
10 | #[error(fmt = core::fmt::LowerHex::fmt)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot have both #[error(transparent)] and #[error(fmt = ...)]
  --> tests/ui/duplicate-fmt.rs:14:1
   |
14 | #[error(fmt = core::fmt::Octal::fmt)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot have both #[error(fmt = ...)] and a format arguments attribute
  --> tests/ui/duplicate-fmt.rs:20:1
   |
20 | #[error("...")]
   | ^^^^^^^^^^^^^^^
